server/models/HostApplication.ts

```diff
@@ -46,17 +46,18 @@ class HostApplication extends Model<InferAttributes<HostApplication>, InferCreat
     host: Collective,
     collective: Collective,
     user: User,
+    status: HostApplicationStatus,
     data: Record<string, unknown>,
   ): Promise<HostApplication> {
-    const existingApplication = await this.getByStatus(host, collective, HostApplicationStatus.PENDING);
+    const existingApplication = await this.getByStatus(host, collective, status);
```

Comment:
On this line, I think we could try to cover an additional use case. If Babel manually applies to OSC (creates pending application) and then applies through the GitHub flow, we should ideally mark the previous, pending application as "Approved" (rather than re-recreating a new one).